src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +270,124 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                                entityName: linkingEntityName,
+                                databaseObject: linkingDbObject,
+                                configEntity: linkingEntity,
+                                entities: new(new Dictionary<string, Entity>()),
+                                rolesAllowedForEntity: new List<string>(),
+                                rolesAllowedForFields: new Dictionary<string, IEnumerable<string>>()
+                            );
+
+                        linkingObjectTypes.Add(linkingEntityName, node);
+                    }
+                }
+            }
+
+            return linkingObjectTypes;
+        }
+
+        /// <summary>
+        /// Helper method to generate object types for linking nodes from (source, target) using
+        /// simple linking nodes which relate the source/target entities with M:N relationship between them.
+        /// </summary>
+        /// <param name="objectTypes">Collection of object types.</param>
+        /// <param name="linkingObjectTypes">Collection of object types for linking entities.</param>
+        /// <param name="entitiesWithManyToManyRelationships">Collection of pair of entities with M:N relationship between them.</param>
+        private void GenerateSourceTargetLinkingObjectDefinitions(
+            Dictionary<string, ObjectTypeDefinitionNode> objectTypes,
+            Dictionary<string, ObjectTypeDefinitionNode> linkingObjectTypes,
+            HashSet<Tuple<string, string>> entitiesWithManyToManyRelationships)
+        {
+            foreach ((string sourceEntityName, string targetEntityName) in entitiesWithManyToManyRelationships)
+            {
+                string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(targetEntityName);
+                ISqlMetadataProvider sqlMetadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                if (sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject? sourceDbo))
+                {
+                    string linkingEntityName = RuntimeConfig.GenerateLinkingEntityName(sourceEntityName, targetEntityName);
+                    ObjectTypeDefinitionNode targetNode = objectTypes[targetEntityName];
+                    IEnumerable<ForeignKeyDefinition> foreignKeyDefinitions = sourceDbo.SourceDefinition.SourceEntityRelationshipMap[sourceEntityName].TargetEntityToFkDefinitionMap[targetEntityName];
+
+                    // Get list of all referencing columns in the linking entity.
+                    List<string> referencingColumnNames = foreignKeyDefinitions.SelectMany(foreignKeyDefinition => foreignKeyDefinition.ReferencingColumns).ToList();
+
+                    NameNode sourceTargetLinkingNodeName = new(LINKING_OBJECT_PREFIX + objectTypes[sourceEntityName].Name.Value + objectTypes[targetEntityName].Name.Value);
```

Comment:
wouldnt the sourceTargetLinkingNodeName's name be same as linkingEntityName?

Reply:
Nope, there are 2 differences:

1. Different prefixes used for linking entity Vs linking node,
2. Linking entity names are solely based on the entity names. While linking node names are generated using the graphql singular name of the entity (if provided) or the entity name.

Reply:
Explained over call.